src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs

```diff
@@ -880,6 +902,164 @@ public void TestMergeChanges_NoMiddleMan()
+            Assert.Equal("new TextChange(new TextSpan(0, 1), (NewLength = 10))", new TextChange(new TextSpan(0, 1), "0123456789").GetDebuggerDisplay());
+        }
+
+        [Fact]
+        [WorkItem(47234, "https://github.com/dotnet/roslyn/issues/47234")]
+        public void Fuzz()
+        {
+            var random = new Random();
+
+            // Adjust upper bound as needed to generate a simpler reproducer for an error scenario
+            var originalText = SourceText.From(string.Join("", Enumerable.Range(0, random.Next(10))));
+
+            foreach (var iteration in Enumerable.Range(0, 100000))
+            {
+                var editedLength = originalText.Length;
+                ArrayBuilder<TextChange> oldChangesBuilder = ArrayBuilder<TextChange>.GetInstance();
+
+                const int maxEditLength = 3;
+                const int maxSkipLength = 2;
+                // generate sequence of "old edits" which meet invariants
+                for (int i = 0; i < originalText.Length; i += 1 + random.Next(maxSkipLength))
+                {
+                    var newText = string.Join("", Enumerable.Repeat('a', random.Next(maxEditLength)));
+                    var newChange = new TextChange(new TextSpan(i, length: random.Next(Math.Min(originalText.Length - i, maxEditLength))), newText);
+                    i = newChange.Span.End;
+
+                    editedLength = editedLength - newChange.Span.Length + newChange.NewText.Length;
+                    oldChangesBuilder.Add(newChange);
+
+                    // Adjust as needed to generate a simpler reproducer for an error scenario
+                    if (oldChangesBuilder.Count == 5) break;
+                }
+
+                var change1 = originalText.WithChanges(oldChangesBuilder);
+
+                ArrayBuilder<TextChange> newChangesBuilder = ArrayBuilder<TextChange>.GetInstance();
+
+                // generate sequence of "new edits" which meet invariants
+                for (int i = 0; i < editedLength; i += 1 + random.Next(maxSkipLength))
+                {
+                    var newText = string.Join("", Enumerable.Repeat('b', random.Next(maxEditLength)));
+                    var newChange = new TextChange(new TextSpan(i, length: random.Next(Math.Min(editedLength - i, maxEditLength))), newText);
+                    i = newChange.Span.End;
+
+                    newChangesBuilder.Add(newChange);
+
+                    // Adjust as needed to generate a simpler reproducer for an error scenario
+                    if (newChangesBuilder.Count == 5) break;
+                }
+
+                var change2 = change1.WithChanges(newChangesBuilder);
+                try
+                {
+                    var textChanges = change2.GetTextChanges(originalText);
+                    Assert.Equal(originalText.WithChanges(textChanges).ToString(), change2.ToString());
+                }
+                catch
+                {
+                    _output.WriteLine($@"
```

Comment:
Love this pattern of having the failure generate another unit test. 

Have you just run this test in a loop on your local box a few thousand to see what it generated?

Reply:
Yes, all new tests named `Fuzz_x` were produced by using this :)

Reply:
Also, this test is enabled to run in CI on an ongoing basis, so we should make sure we have high confidence in its reliability.